### PR TITLE
fix(react-virtualizer): fix errors when the items order changes

### LINKED ISSUE #526: [Bug](@fluentui-contrib/react-virtualizer): Re-render reconciliation fails with unique key errors when the items order changes (starting with version 0.4.0)
### Package

azure-theme

### Package version

0.5.2

### Environment

```shell
Local:
  System:
    OS: Windows 11 10.0.26200
    CPU: (24) x64 13th Gen Intel(R) Core(TM) i7-13700
    Memory: 25.25 GB / 63.64 GB
  Browsers:
    Chrome: 141.0.7390.66
    Edge: Chromium (141.0.3537.71)
    Firefox: 144.0 - C:\Program Files\Mozilla Firefox\firefox.exe
    Internet Explorer: 11.0.26100.1882

Stackblitz:
  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  npmPackages:
    @fluentui-contrib/react-virtualizer: ^0.5.1 => 0.5.2 
    @types/react: ^19.2.2 => 19.2.2 
    @types/react-dom: ^19.2.2 => 19.2.2 
    react: ^19.2.0 => 19.2.0 
    react-dom: ^19.2.0 => 19.2.0
```

### Current Behavior

When items are re-rendered and their order has changed, React reconciliation fails with errors about keys not being unique. This behavior started in version 0.4.0 and still occurs in the latest (as of logging this bug) 0.5.2.

### Expected Behavior

If the items order changes and keys are still unique, React reconciliation should succeed.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-1yshvhpv?file=src%2FApp.tsx

### Steps to reproduce

1. Click the "count is 0" button
2. Click the "count is 1" button
3. Observe the incorrectly rendered list of items and the errors in the console

### Are you reporting an Accessibility issue?

no

### Suggested severity

High - No workaround

### Products/sites affected

Babylon.js tools

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by dmytrokirpa**
This PR fixes item order issues in `react-virtualizer`, resolving React key errors and improving stability. It also adds thorough tests for edge cases.

Fixes #526

### CODE REVIEW

File: packages/react-virtualizer/src/components/Virtualizer/Virtualizer.component-browser-spec.tsx [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    await expect(item1).toContainText('Item 1');
+  });
+
+  test('should handle edge case at list boundaries', async ({ mount }) => {
+    const component = await mount(<VirtualizerExample numItems={100} />);
+    const container = component.getByTestId('scroll-container');
+
+    // Scroll to very bottom
+    await container.evaluate((el) => {
+      el.scrollTop = el.scrollHeight - el.clientHeight;
```
Comment (mainframev) [inline packages/react-virtualizer/src/components/Virtualizer/Virtualizer.component-browser-spec.tsx]: should we also dispatch a scroll event, wdyt? 

```suggestion
      el.scrollTop = el.scrollHeight - el.clientHeight;\
      el.dispatchEvent(new Event('scroll'));
```

File: packages/react-virtualizer/src/components/Virtualizer/Virtualizer.component-browser-spec.tsx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    await expect(newFirstItem).toBeVisible({ timeout: 3000 });
+    await expect(newFirstItem).toHaveAttribute('data-value', '10');
+
+    // No key reconciliation errors
+    const keyErrors = consoleErrors.filter(
+      (err) =>
+        err.includes('key') ||
+        err.includes('unique') ||
+        err.includes('Warning: Encountered two children with the same key')
+    );
```
Comment (mainframev) [inline packages/react-virtualizer/src/components/Virtualizer/Virtualizer.component-browser-spec.tsx]: even if it's a test, you can make it more DRY and descriptive 

make a function outside of the tests, something as this: 

```ts
const hasKeyErrors = (errors: string[]) => {
    const regex = /(key|unique|Encountered two children with the same key)/i;
    return errors.some(err => regex.test(err));
};

// test
expect(hasKeyErrors(consoleErrors)).toBe(false);
```

**Review by mainframev — APPROVED** [review]
